plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -193,11 +194,37 @@ export class UserInteractionPlugin extends BasePlugin<unknown> {
             return listener.apply(target, args);
           }
         };
+        plugin._wrappedListeners.set(listener, patchedListener);
```

Comment:
It looks like the listeners are assumed to be unique. If so this wouldn't handle reusing the same listener, though I don't know if that's an edge case we want to worry about.

Reply:
No, this is an important point, thanks.  In particular I'm going to tweak it to properly support
```
  element.addEventListener('type1', listener);
  element.addEventListener('type2', listener);
  element.removeEventListener('type1', listener);
```
Which the current instrumentation does not properly support.  Thanks for the careful review!

Reply:
When I start thinking more about this it is becoming more complicated when I originally thought and it will require to make few other changes because when you call AddEventListener with the same callback and type it will overwrite previous callback. Also it should only create span for click event. So when saving patchedListener you should also save the type

If I haven't missed anything the patch function should look like
```
  private _wrappedListeners = new WeakMap<Function, Record<string, ()=>{}>>();
  private _patchElement() {
    const plugin = this;
    return (original: Function) => {
      return function addEventListenerPatched(
        this: HTMLElement,
        type: string,
        listener: any,
        useCapture: any
      ) {
        const patchedListener = (...args: any[]) => {
          const target = this;
          if (type.toLowerCase() !== 'click') {
            return listener.apply(target, args);
          }
          const span = plugin._createSpan(target, 'click');
          if (span) {
            return plugin._tracer.withSpan(span, () => {
              const result = listener.apply(target, args);
              // no zone so end span immediately
              span.end();
              return result;
            });
          } else {
            return listener.apply(target, args);
          }
        };
        let listeners = plugin._wrappedListeners.get(original);
        if (!listeners) {
          listeners = {};
          listeners[type] = patchedListener;
          plugin._wrappedListeners.set(original, listeners);
        } else if (listeners[type]) {
          this.removeEventListener(type, listeners[type]);
          delete listeners[type];
        }
        listeners[type] = patchedListener;
        // not sure if the below line is yet needed
        plugin._wrappedListeners.set(original, listeners);
        return original.call(this, type, patchedListener, useCapture);
      };
    };
  }
```

After this you will also need to modify the patch for `removeEventListener` so that it will take into consideration the type.

Reply:
This is very similar to where I landed, with the added bit that, of course, the element it's on is a part of the registration.  I've got a separate patch for better handling of the event type and would rather not conflate the two.